File: pyworks/ch05/passed_time.py
# 지나온 날짜 계산하기 : 일수 = 종료일 - 시작일
# datetime.date(종료일) - datetime.date(시작일)
# 요일(인덱스) - datetime.date(특정일).weekday(), 월-0번
import datetime
import calendar

print(" ♠ 지금까지 몇 일? ♠")
day1 = datetime.date(2023, 3, 16)
print(day1)

# today = datetime.date(2023, 4, 21)
today = datetime.date.today()
print(today)

passed_time = today - day1
#print(passed_time)
# 날짜객체.days -> 일로 환산
print("개강 이후 {}일이 지났습니다.".format(passed_time.days))

# 날짜로 요일 알아내기
days = ['월', '화', '수', '목', '금', '토', '일']

# datetime.date(2023, 3, 16).weekday()
the_day = datetime.date(2023, 3, 16).weekday()
print(the_day)
print(f'{days[the_day]}요일')

now = datetime.date.today().weekday()
print(f'{days[now]}요일')

# 2023. 3월 출력
calendar.prcal(2023)
calendar.prmonth(2023, 3)

File: javaworks/ch03/src/datetime/PassedTime.java
package datetime;

import java.util.Calendar;

public class PassedTime {

	public static void main(String[] args) {
		Calendar theDay = Calendar.getInstance();  //시작일 객체 생성
		Calendar today = Calendar.getInstance();   //오늘 객체 생성
		
		//System.out.println(today);
		//System.out.println(today.get(Calendar.YEAR)); 
		//System.out.println(today.get(Calendar.MONTH) + 1);
		
		//System.out.println(today.getTimeInMillis());
		//년으로 환산
		//System.out.println(today.getTimeInMillis() / (24*60*60*1000)/ 365);
		//일로 환산
		//System.out.println(today.getTimeInMillis() / (24*60*60*1000));
		
		//시작일과 종료일 지정
		theDay.set(2023, 3, 16);  //시작일 지정
		today.set(2023, 5, 24);   //종료일 지정
		
		//날짜 출력
		/*String startDay = theDay.get(Calendar.YEAR) + "년 " +
			theDay.get(Calendar.MONTH) + "월 " + theDay.get(Calendar.DATE) + "일";
		String today1 = today.get(Calendar.YEAR) + "년 " +
				today.get(Calendar.MONTH) + "월 " + today.get(Calendar.DATE) + "일";
		System.out.println(startDay);
		System.out.println(today1);*/
		
		//날짜 출력 메서드 호출
		System.out.println(datePrint(theDay)); //시작일 출력
		System.out.println(datePrint(today));  //종료일 출력
		
		//지나온 시간 계산 : 종료일 - 시작일
		long passedTime = today.getTimeInMillis() - theDay.getTimeInMillis();
		System.out.println(passedTime + "밀리초");
		//일로 환산
		passedTime = passedTime/(24*60*60*1000);
		System.out.println("개강 이후 " + passedTime + "일이 지났습니다.");
	}//main 끝
	
	//날짜 출력 메서드
	public static String datePrint(Calendar date) { //날짜 객체를 매개로 전달받아
		String strDate = date.get(Calendar.YEAR) + "년 " +
				date.get(Calendar.MONTH) + "월 " + date.get(Calendar.DATE) + "일";
		return strDate;  //날짜 문자열을 반환(돌려줌)
	}
}
